include/aero/array/array.hpp

```diff
@@ -11,35 +14,41 @@ class Array {
   Array(const Array& other) = default;
   /// Move constructor
   Array(Array&& other) = default;
+  /// Constructor from dimensions. Array elements
+  /// are initialized to zero.
+  Array(const unsigned int number_of_elements);
```

Comment:
It's customary in C++ (and C, actually) to use `size_t` instead of `unsigned int` (in C++, it's pronounced `std::size_t` because the pedants have won the war there). Also, for scalar types like integers, `const` isn't really all that useful. `const` is another topic of great controversy in C++, because it tends to propagate itself throughout the contexts from which it's used, which some of us feel is an anti-pattern. This is another topic we'll probably be talking about. :-)

Reply:
Also, another silly C++ thing: single-argument constructors aside from the copy and move constructors need the `explicit` keyword in front of them to prevent implicit casting from occurring. In other words, with this constructor (and the `vector` one below), I could create an `Array` with

```
Array a = 6;
```

Personally, I think this is a huge mistake the C++ people made early one, so I always use `explicit`. These kinds of situations ("Dear compiler, please don't do the stupid thing you do by default--here's a keyword for you...") are common in C++ and make it a very tricky language to work in.